src/categorical_algebra/CatElements.jl

```diff
@@ -0,0 +1,82 @@
+module CatElements
+using Catlab
+using ..CSets
+using ..FinSets
+using ...Present, ...Theories
+# using ..CSetDataStructures
+using ...Theories: Category, CatDesc, AttrDesc, ob, hom, attr, adom, acodom
+import Catlab.Present: Presentation
+
+export ThElements, AbstractElements, Elements, elements
+
+
+@present ThElements(FreeSchema) begin
+    (El, Arr, Ob, Hom)::Ob
+    Name::Data
+    src::Hom(Arr, El)
+    tgt::Hom(Arr, El)
+    dom::Hom(Hom, Ob)
+    cod::Hom(Hom, Ob)
+    πₑ::Hom(El, Ob)
+    πₐ::Hom(Arr, Hom)
+    nameo::Attr(Ob, Name)
+    nameh::Attr(Hom, Name)
+end
+
+const AbstractElements = AbstractACSetType(ThElements)
+const Elements = ACSetType(ThElements, index=[:src, :tgt, :πₑ, :πₐ])
+
+"""    elements(X::AbstractACSet)
+
+construct the category of elements from an ACSet. This only correctly handles the CSet part.
+This transformation converts an instance of C into a Graph homomorphism. The codomain of the
+homomorphism is a graph shaped like the schema. This is one half of the isomorphism between
+databases and knowledge graphs.
+"""
+function elements(X::ACS) where
```

Comment:
In theory, we should have conversion between schemas and graphs implemented generically in another place, and this should use that, but we can do that in another PR if you don't want to do that here.